### PR TITLE
Ensure kv collection refresh settings are not considered unless the feature is enabled.

### PR DESCRIPTION
**Pull request by jimmyca15**
This PR does the following:

* Ensure that kv collection refresh interval is not used unless collection based refresh of key-values is enabled.
* Fix short circuit code path when no refresh is due.
* Add tests to ensure that minimum refresh interval is respected for key-values and feature flags.

Impact wise, these changes should have little to no behavioral/performance impact. Mostly code clarity.

In response to https://github.com/Azure/AppConfiguration-DotnetProvider/issues/632

### CODE REVIEW

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs [hunk 1]
```diff
@@ -118,6 +118,13 @@ public AzureAppConfigurationProvider(IConfigurationClientManager configClientMan
 
             if (options.RegisterAllEnabled)
             {
+                if (options.KvCollectionRefreshInterval <= TimeSpan.Zero)
```
Comment (amerjusupovic) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs:121]: Just wanted to mention that this shouldn't be possible based on how the refresh interval is set by default if `ConfigureRefresh` and `RegisterAll` were called, but this is fine if we want to just check here for the future

**Comment by jimmyca15** [comment]
@amerjusupovic it looks like the tests i added already existed. I removed them. But I noticed that tests were using Thread.Sleep, I replaced them with Task.Delay.

**Review by amerjusupovic — APPROVED** [review]
